server/mediaserver.go

```diff
@@ -550,7 +550,7 @@ func (s *LivepeerServer) registerConnection(ctx context.Context, rtmpStrm stream
 	if exists {
 		// We can only have one concurrent stream per ManifestID
 		s.connectionLock.Unlock()
-		cxn.sessManager.cleanup(nil)
+		cxn.sessManager.cleanup(context.Background())
```

Comment:
Is there a way to update [this test](https://github.com/livepeer/go-livepeer/blob/master/server/mediaserver_test.go#L1157) so that without this change the test would trigger a panic?

Reply:
Tagging in @cyberj0g as this line was last updated in a1fb76147.

FWIW I took a quick look at the [linked test](https://github.com/livepeer/go-livepeer/blob/master/server/mediaserver_test.go#L1157) and it seems like it'd be pretty tricky to trigger the code path that would result in this cleanup() call since registerConnection() will return `errAlreadyExists` in an [initial connection existence check](https://github.com/livepeer/go-livepeer/blob/3ff79bd7045fc72c1d810736a17af6128b83905b/server/mediaserver.go#L522) prior to the [second existence check](https://github.com/livepeer/go-livepeer/blob/3ff79bd7045fc72c1d810736a17af6128b83905b/server/mediaserver.go#L550) that is used for the clause that contains the cleanup() call. The existing [concurrent HandlePush test](https://github.com/livepeer/go-livepeer/blob/3ff79bd7045fc72c1d810736a17af6128b83905b/server/push_test.go#L1429) might be able to trigger this code path with some tweaking, but seems like that might be potentially flaky as you would still lack a way to deterministically trigger the clause for that second connection existence check...Worth another look, but if it a substantial effort then I think it is fine to merge this as-is (could consider refactoring at some point to make this type of test easier).